File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        path = []
        ans  = []
        def dfs(nums: List[int], start: int) -> None:
            ans.append(path[:])
            if start == n:
                return 
            for i in range(start,n):
                path.append(nums[i])
                dfs(nums,i+1)
                path.remove(nums[i])
        dfs(nums,0)
        return ans

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<int> path;
    void dfs(const vector<int>& nums, int n, int start){
        ans.push_back(path);
        if (start == n)
            return;
        for (int i=start; i<n; i++){
            path.push_back(nums[i]);
            dfs(nums,n,i+1);
            path.pop_back();
        }
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        int n = (int)nums.size();
        dfs(nums,n,0);
        return ans;
    }
};